lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,13 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        if e.message =~ /8007089A/m
```

Comment:
Let's do that. DO we get any other type of error? Or should I say, should we capture any other type of error?